src/main/java/com/alex/eshop/eshop/Service/CategoryServiceImpl.java

```diff
@@ -0,0 +1,37 @@
+package com.alex.eshop.eshop.Service;
+
+import com.alex.eshop.eshop.Entity.Category;
+import com.alex.eshop.eshop.ExceptionHandling.NoSuchCategoryException;
+import com.alex.eshop.eshop.Repository.CategoryRepository;
+import org.springframework.stereotype.Component;
+import org.springframework.stereotype.Service;
+
+import javax.transaction.Transactional;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@Transactional
+public class CategoryServiceImpl implements CategoryService{
+    private final CategoryRepository categoryRepository;
+    public CategoryServiceImpl(CategoryRepository categoryRepository){
+        this.categoryRepository = categoryRepository;
+    }
+    @Override
+    public List<Category> getAllCategories() {
+        return categoryRepository.findAll();
+    }
+    public Category getCategory(Long id){
```

Comment:
Function `public Item getItemWithCategoryInfo(Long id);` is good, but it can be improved. **Code can be reduced to 1-2 lines** if use orElseThrow() for Optional.

```
return categoryRepository.findById(id)
        .orElseThrow(() -> new NoSuchCategoryException("There is no category with id " + id));
```